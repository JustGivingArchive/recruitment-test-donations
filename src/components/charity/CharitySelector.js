import React from 'react';
import IndividualCharity from './IndividualCharity.js';

// donations are served from the API as an array, so we can map over them
// and create a donation component for each

const CharitySelector = (props) => {
  const charitys = [{
    name: 'British Heart Foundation',
    id: '183092'
  },
  {
    name: 'Cancer Research UK',
    id: '2357'
  },
  {
    name: 'Macmillan Cancer Support',
    id: '2116'
  },
  {
    name: 'Oxfam',
    id: '13441'
  },
  {
    name: 'National Trust',
    id: '183560'
  },
  {
    name: 'Save the Children',
    id: '18570'
  }
  ];
  return (
    <div className="content_container">
      {charitys.map((charity) => <IndividualCharity select={props.select} {...charity} />)}
    </div>
  );
};

export default CharitySelector;
