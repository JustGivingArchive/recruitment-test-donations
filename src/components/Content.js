import React from 'react';
import Charity from './charity/Charity.js';
import Donations from './donations/Donations.js';
import CharitySelector from './charity/Charityselector.js';

const Content = (props) => {
  console.log(props);
  if (props.dropDown) {
    return (
      <CharitySelector select={props.select} />
    )
  }

  if (props.charityLoaded && props.donationsLoaded) {
    return (
      <div>
        <Charity info={props.charityInfo} />
        <Donations donations={props.donations} />
      </div>
    );
  }

  return (
      <div className="main_container">
        <h2 className="subtitle">Loading</h2>
      </div>
  );
};

export default Content;
