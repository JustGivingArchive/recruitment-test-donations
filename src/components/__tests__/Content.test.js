import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import Content from '../Content.js';
import CharitySelector from '../charity/CharitySelector.js';

describe('Test content component', () => {
  let state;
  let mounted;
  const content = () => {
    if (!mounted) {
      mounted = mount(
        <Content
          {...state}
          toggle={ () => null }
          select={ () => null }
        />
      );
    }
    return mounted;
  }

  beforeEach(() => {
    state = {
      charityInfo: null,
      donations: null,
      charityLoaded: false,
      donationsLoaded: false,
      dropDown: true
    };
    mounted = undefined;
  });

  it('matches snapshot', () => {
    const tree = renderer.create(
      <Content
        {...state}
        toggle={ () => null }
        select={ () => null }
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("wrapping div contains everything", () => {
    const divs = content().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(content().children());
  });
});
