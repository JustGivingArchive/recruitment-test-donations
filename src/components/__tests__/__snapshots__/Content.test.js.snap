exports[`Test content component matches snapshot 1`] = `
<div>
  <div
    className="main_container">
    <h2
      className="subtitle">
      Please select a charity to see donations.
    </h2>
  </div>
  <div
    className="content_container">
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        British Heart Foundation
      </p>
    </div>
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        Cancer Research UK
      </p>
    </div>
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        Macmillan Cancer Support
      </p>
    </div>
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        Oxfam
      </p>
    </div>
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        National Trust
      </p>
    </div>
    <div
      className="panel hoverable"
      onClick={[Function]}>
      <p
        className="paneltextlarge">
        Save the Children
      </p>
    </div>
  </div>
</div>
`;
